treq/client.py

```diff
@@ -195,6 +198,13 @@ def request(self, method, url, **kwargs):
                     b'content-type', [b'application/x-www-form-urlencoded'])
                 data = urlencode(data, doseq=True)
             bodyProducer = self._data_to_body_producer(data)
+        elif json:
+            # If data is sent as json, set Content-Type as 'application/json'
+
+            headers.setRawHeaders(
+                b'content-type', [b'application/json'])
+            json = json_dumps(json).encode('ascii')
```

Comment:
Could you only do the encode `if _PY3`?  Also please don't reassign `json` with a different type; that could make debugging confusing.  Call it `json_string` or something.